js/src/Zone.js

```diff
@@ -0,0 +1,57 @@
+import ExpectedMark from "./ExpectedMark";
+import UXCapture from "./UXCapture";
+
+// array holding current, global set of expected zones
+let expectedZones = [];
+
+/**
+ * Zone represents collection of elements groupped together and corresponding phase of page load
+ *
+ * Only one set of zones can be tracked for the same view at one point in time
+ */
+export default class Zone {
+  static setExpectedZones(zones) {
+    expectedZones = zones.map(zone => {
+      // only create promises if zone contains any marks, otherwise just ignore it
+      if (zone.marks && zone.marks.length > 0) {
+        const expectedZone = new Zone(zone.label, zone.marks);
+
+        return expectedZone;
+      }
+    });
+  }
+
+  // constructs individual zone object
+  constructor(label, marks, elements = []) {
```

Comment:
let's rename `marks` to `markLabels` just to make it clear it is not a collection of `Mark` instances.